src/server/main_service.cc

```diff
@@ -695,9 +695,23 @@ bool Service::VerifyCommand(const CommandId* cid, CmdArgList args, ConnectionCon
 
   bool is_trans_cmd = CO::IsTransKind(cid->name());
   bool under_script = bool(dfly_cntx->conn_state.script_info);
-  bool blocked_by_loading = !dfly_cntx->journal_emulated && etl.gstate() == GlobalState::LOADING &&
-                            (cid->opt_mask() & CO::LOADING) == 0;
-  if (blocked_by_loading || etl.gstate() == GlobalState::SHUTTING_DOWN) {
+  bool allowed_by_state = true;
+  switch (etl.gstate()) {
+    case GlobalState::LOADING:
+      allowed_by_state = dfly_cntx->journal_emulated || (cid->opt_mask() & CO::LOADING);
+      break;
+    case GlobalState::SHUTTING_DOWN:
+      allowed_by_state = false;
+      break;
+    case GlobalState::TAKEN_OVER:
+      allowed_by_state = cid->name() == "REPLCONF" || cid->name() == "SAVE";
```

Comment:
why do you return error instead of blocking the command till takeover is done?

Reply:
Without proxy support I don't think there's much of a difference between blocking and denying, but if we want to go for it I'd rather do it in a different PR

Reply:
If the replica takeover failed f.e due to timeout in this implementation you are rejecting commands instead of waiting for the takeover to fail and then the commands will resume.
You are suggesting a different PR, but doesnt the transaction guard you are using solves this issue already?

Reply:
It doesn't work - some commands will get executed despite the transaction guard - but I need to think about this a bit and understand why :/

Reply:
I believe this is either commands which do not use the transaction framework which I believe we don't care about to be executed as they should not change the database data or the inline scheduling which we can be turned off

Reply:
So, my problem is basically this:

1. We receive commands during the take over and schedule them into `TxQueue`.
2. They get executed when the `TransactionGuard` is released, which can happen because I do the shutdown async.

I think we could maybe flush the transaction queues and cancel the commands there, but I don't know if that's possible? Other than that, what I implemented is to just preemptively prevent commands from getting to the `TxQueue` with this.

[take_over_guard.txt](https://github.com/dragonflydb/dragonfly/files/11880250/take_over_guard.txt)